src/lib/kadm5/admin.h

```diff
@@ -455,16 +467,11 @@ krb5_error_code kadm5_init_krb5_context (krb5_context *);
 
 krb5_error_code kadm5_init_iprop(void *server_handle, char **db_args);
 
-/*
- * kadm5_get_principal_keys is used only by kadmin.local to extract existing
- * keys from the database without changing them.  It should never be exposed
- * to the network protocol.
- */
 kadm5_ret_t    kadm5_get_principal_keys(void *server_handle,
                                         krb5_principal principal,
-                                        krb5_keyblock **keyblocks,
-                                        int *n_keys);
-
+                                        krb5_kvno kvno,
```

Comment:
You can't do this.  `kadm5_get_principal_keys()` has been in `src/lib/kadm5/srv/libkadm5srv_mit.exports` since 2010.  Wallet and who knows what else might be using this function.  You'll need a new one.